opentelemetry/src/metrics/instruments/mod.rs

```diff
@@ -87,7 +87,26 @@ impl<'a, T> HistogramBuilder<'a, T> {
     ///
     /// Setting boundaries is optional. By default, the boundaries are set to:
     ///
-    /// `[0.0, 5.0, 10.0, 25.0, 50.0, 75.0, 100.0, 250.0, 500.0, 750.0, 1000.0, 2500.0, 5000.0, 7500.0, 10000.0]`
+    /// `[0.0, 5.0, 10.0, 25.0, 50.0, 75.0, 100.0, 250.0, 500.0, 750.0, 1000.0,
+    /// 2500.0, 5000.0, 7500.0, 10000.0]`
+    ///
+    /// # Notes
+    /// - Boundaries must not contain `f64::NAN`, `f64::INFINITY` or
+    ///   `f64::NEG_INFINITY`
+    /// - Values must be in strictly increasing order (e.g., each value must be
+    ///   greater than the previous).
+    /// - Boundaries must not contain duplicate values.
+    ///
+    /// If invalid boundaries are provided, the instrument will not report
+    /// measurements.
+    /// Providing an empty `vec![]` means no bucket information will be
```

Comment:
With the current implementation, we would still count and export the bucket information. It would just be one bucket though. We should probably change the implementation to avoid bucket counting altogether in case of empty bounds.

Reply:
https://github.com/open-telemetry/opentelemetry-rust/issues/2354